tracer/src/Datadog.Trace/Agent/Transports/ApiWebRequest.cs

```diff
@@ -46,5 +48,71 @@ public async Task<IApiResponse> PostAsync(ArraySegment<byte> bytes, string conte
                 return new ApiWebResponse((HttpWebResponse)exception.Response);
             }
         }
+
+        public async Task<IApiResponse> PostAsync(params MultipartFormItem[] items)
+        {
+            if (items is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(items));
+            }
+
+            Log.Debug<int>("Sending multipart form request with {Count} items.", items.Length);
+
+            var boundary = "---------------------------" + DateTime.Now.Ticks.ToString("x");
```

Comment:
nit: just wondering can we drop an allocation here by splitting `boundary`?

e.g.

```csharp
const string boundary = "---------------------------";
string nowTicks = DateTime.Now.Ticks.ToString("x");

// ...

// compiler will merge the first two constants         👇
_request.ContentType = "multipart/form-data; boundary=" + boundary + nowTicks;

// ...

// compiler will merge the first two constants      👇
var boundaryBytes = Encoding.ASCII.GetBytes("\r\n--" + boundary + nowTicks + "\r\n");
```

Maybe an over optimisation/already handled by the compiler/no real benefit, just wondering 🙂

Reply:
made similar change in [84ecdc7](https://github.com/DataDog/dd-trace-dotnet/pull/2962/commits/84ecdc7a6638c4593e83777456e9370644539de4)